src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(FigureList.values().length);
+        String figureName = FigureList.values()[index].name();
+
+        Color randomColor = Color.valueOf(colorSupplier.getRandomColor());
+
+        Figure randomFigure = null;
```

Comment:
Returning `null` from a method is a bad practice. In the `default` case of the switch statement, you can directly return the new `Square` instance.